quantmsutils/diann/diann2mztab.py

```
@@ -311,8 +311,8 @@ def compute_mass_modified_peptide(peptide_seq: str) -> float:
 class DiannDirectory:
     def __init__(self, base_path, diann_version_file):
         self.base_path = Path(base_path)
-        if not self.base_path.exists() and not self.base_path.is_dir():
-            raise NotADirectoryError(f"Path {self.base_path} does not exist")
+        if not self.base_path.exists() or not self.base_path.is_dir():
+            raise NotADirectoryError(f"Path {self.base_path} does not exist or is not a directory")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)